src/test/java/dev/openfeature/sdk/EventsTest.java

```diff
@@ -594,16 +596,18 @@ void mustPersistAcrossChanges() {
 
     @Nested
     class HandlerRemoval {
-        @Specification(number="5.2.7", text="The API and client MUST provide a function allowing the removal of event handlers.")
+        @Specification(number = "5.2.7", text = "The API and client MUST provide a function allowing the removal of event handlers.")
         @Test
         @DisplayName("should not run removed events")
+        @SneakyThrows
         void removedEventsShouldNotRun() {
             final String name = "removedEventsShouldNotRun";
             final Consumer<EventDetails> handler1 = mockHandler();
             final Consumer<EventDetails> handler2 = mockHandler();
 
             TestEventsProvider provider = new TestEventsProvider(INIT_DELAY);
-            OpenFeatureAPI.getInstance().setProvider(name, provider);
+            provider.initialize(null);
+            OpenFeatureAPI.getInstance().setProviderAndWait(name, provider);
```

Comment:
```suggestion
            OpenFeatureAPI.getInstance().setProviderAndWait(name, provider);
```

At least with your latest changes this works fine. I can't make sense of the manual init call.